cdap-api/src/main/java/co/cask/cdap/api/service/http/HttpServiceResponder.java

```diff
@@ -139,11 +142,76 @@
   /**
    * Sends response back to client.
    *
-   * @param status status of the response
+   * @param status status of the HTTP response
    * @param content content to be sent back
    * @param contentType type of content
    * @param headers headers to send; each {@link Map.Entry} contains the header name and value to be sent, hence
    *                multiple values for the same header name is allowed.
    */
   void send(int status, ByteBuffer content, String contentType, Iterable<? extends Map.Entry<String, String>> headers);
+
+  /**
+   * Sends response back to client with response body produced by the given {@link HttpContentProducer}.
+   *
+   * @param status status of the HTTP response
+   * @param producer a {@link HttpContentProducer} to produce content to be sent back
+   * @param contentType type of content
+   */
+  void send(int status, HttpContentProducer producer, String contentType);
```

Comment:
why is it that the content length is given by the producer itself, whereas the content type is a separate parameter?